File: Priprema K3 - Predavanje/IN/Python/teretni_voz.py
'''
Created on Dec 25, 2018

@author: Marko
'''

from vagon import Vagon
from roba import Roba

class TeretniVoz:

    def __init__(self, brVagona, maxDozvoljenoKG):
        self.vagoni = []
        while brVagona > 0:
            self.vagoni.append(Vagon())
            brVagona -= 1
        self.vagonMaxKG = maxDozvoljenoKG
        
    def getSlobodanProstor(self, brVagona):
        if type(brVagona) is not int: #NE MORA OVA PROVERA
            return -1
        if brVagona < 0 or brVagona >= len(self.vagoni):
            return -1
        return self.vagonMaxKG - self.vagoni[brVagona].getTezinaRobe()
    
    def utovariTeret(self, teret, brVagona):
        if type(teret) is not Roba: #NE MORA OVA PROVERA
            return False
        if type(brVagona) is not int: #NE MORA OVA PROVERA
            return False
        
        if brVagona < 0 or brVagona >= len(self.vagoni):
            return False
        if teret.tezinaKG > self.getSlobodanProstor(brVagona):
            return False
        self.vagoni[brVagona].dodajRobu(teret)
        return True
    
    def getMarkaNajtezeRobe(self):
        kgNajteze = 0
        markaNajteze = "PRAZAN"
        for v in self.vagoni:
            for r in v.teret:
                if r.tezinaKG > kgNajteze:
                    markaNajteze = r.marka
                    kgNajteze = r.tezinaKG
        return markaNajteze
        
    def __str__(self):
        ret = "TERETNI VOZ\n"
        ret += "Maksimalna tezina po vagonu = " + str(self.vagonMaxKG) + " KG\n"
        for v in self.vagoni:
            ret += "---------------------------------------\n"
            ret += str(v)
            ret += "\n---------------------------------------\n"
        return ret

File: Priprema K3 - Predavanje/IN/Java/TeretniVoz.java
public class TeretniVoz {
	private Vagon[] vagoni;
	private int vagonMaxKG;
	
	public TeretniVoz(int brVagona, int maxDozvoljenoKG) {
		vagoni = new Vagon[brVagona];
		for(int i = 0; i < vagoni.length; ++i) {
			vagoni[i] = new Vagon();
		}
		vagonMaxKG = maxDozvoljenoKG;
	}
	
	public int getSlobodanProstor(int brVagona) {
		if(brVagona < 0 || brVagona >= vagoni.length) {
			return -1;
		}
	
		return vagonMaxKG - vagoni[brVagona].getTezinaRobe();
	}
	
	public boolean utovariTeret(Roba teret, int brVagona) {
		if (brVagona < 0 || brVagona >= vagoni.length) {
			return false;
		}
		
		if (teret.getTezinaKG() > this.getSlobodanProstor(brVagona)) {
			return false;
		}
		
		vagoni[brVagona].dodajRobu(teret);
        return true;
	}
    
	public String getMarkaNajtezeRobe() {
		int kgNajteze = 0;
		String markaNajteze = "PRAZAN";
		for(int i = 0; i < vagoni.length; ++i) {
			for(int j = 0; j < vagoni[i].getTeret().size(); ++j) {
				if(vagoni[i].getTeret().get(j).getTezinaKG() > kgNajteze) {
					markaNajteze = vagoni[i].getTeret().get(j).getMarka();
					kgNajteze = vagoni[i].getTeret().get(j).getTezinaKG();
				}
			}
		}
				
        return markaNajteze;
	}
	
	@Override
	public String toString() {
		String ret = "TERETNI VOZ\n";
        ret += "Maksimalna tezina po vagonu = " + vagonMaxKG + " KG\n";
        for(int i = 0; i < vagoni.length; ++i) {
            ret += "---------------------------------------\n";
            ret += vagoni[i];
            ret += "\n---------------------------------------\n";
        }
        return ret;
	}
}